zebra-chain/src/block/tests/vectors.rs

```diff
@@ -145,136 +142,119 @@ fn block_limits_single_tx() {
-        .expect("2 hours and 1 second in the past should be valid as a block header time");
-}
-
-/// Valid unix epoch timestamps for blocks, in seconds
-static BLOCK_HEADER_VALID_TIMESTAMPS: &[i64] = &[
-    // These times are currently invalid DateTimes, but they could
-    // become valid in future chrono versions
-    i64::MIN,
-    i64::MIN + 1,
-    // These times are valid DateTimes
-    (i32::MIN as i64) - 1,
-    (i32::MIN as i64),
-    (i32::MIN as i64) + 1,
-    -1,
-    0,
-    1,
-    LockTime::MIN_TIMESTAMP - 1,
-    LockTime::MIN_TIMESTAMP,
-    LockTime::MIN_TIMESTAMP + 1,
-];
-
-/// Invalid unix epoch timestamps for blocks, in seconds
-static BLOCK_HEADER_INVALID_TIMESTAMPS: &[i64] = &[
-    (i32::MAX as i64) - 1,
-    (i32::MAX as i64),
-    (i32::MAX as i64) + 1,
-    LockTime::MAX_TIMESTAMP - 1,
-    LockTime::MAX_TIMESTAMP,
-    LockTime::MAX_TIMESTAMP + 1,
-    // These times are currently invalid DateTimes, but they could
-    // become valid in future chrono versions
-    i64::MAX - 1,
-    i64::MAX,
-];
-
-#[test]
-fn time_check_fixed() {
-    // These checks are non-deterministic, but the times are all in the
-    // distant past or far future. So it's unlikely that the test
-    // machine will have a clock that makes these tests fail.
-    let now = Utc::now();
-
-    for valid_timestamp in BLOCK_HEADER_VALID_TIMESTAMPS {
-        let block_header_time = match Utc.timestamp_opt(*valid_timestamp, 0) {
-            LocalResult::Single(time) => time,
-            LocalResult::None => {
-                // Skip the test if the timestamp is invalid
-                continue;
-            }
-            LocalResult::Ambiguous(_, _) => {
-                // Utc doesn't have ambiguous times
-                unreachable!();
-            }
-        };
-        node_time_check(block_header_time, now)
-            .expect("the time should be valid as a block header time");
-        // Invert the check, leading to an invalid time
-        node_time_check(now, block_header_time)
-            .expect_err("the inverse comparison should be invalid");
-    }
-
-    for invalid_timestamp in BLOCK_HEADER_INVALID_TIMESTAMPS {
-        let block_header_time = match Utc.timestamp_opt(*invalid_timestamp, 0) {
-            LocalResult::Single(time) => time,
-            LocalResult::None => {
-                // Skip the test if the timestamp is invalid
-                continue;
-            }
-            LocalResult::Ambiguous(_, _) => {
-                // Utc doesn't have ambiguous times
-                unreachable!();
-            }
-        };
-        node_time_check(block_header_time, now)
-            .expect_err("the time should be invalid as a block header time");
-        // Invert the check, leading to a valid time
-        node_time_check(now, block_header_time).expect("the inverse comparison should be valid");
-    }
-}
+// TODO Move these tests to zebra-consensus
```

Comment:
@vramana do you need help moving these tests to zebra-consensus?

We usually don't merge PRs that remove tests - the tests help us make sure that the changes in the PR actually work.

Reply:
That makes sense.

But I think @hdevalence meant that we would split the tests, leave the dependent tests in zebra-chain, and leave both sets of tests enabled.

Because all the tests are enabled, we know that the code still works. But we also split the tests, so we aren't blocking the PR on the test refactoring.

I opened #1021 as a follow-up ticket, so we don't forget to do this extra work.

What do you think?

Reply:
Looks good. So this PR will get merged with these commented tests right?

Reply:
my interpretation of @teor2345's comment is that you should still uncomment the tests but I might be misinterpreting.

Reply:
But they won't compile.

Reply:
aah, I misunderstood, I thought this was moving code from consensus to chain not the other way, my bad for not actually reviewing before chiming in. 

I don't see how we can fix these to compile without moving back the header checks so this is good I think and you interpreted their comments correctly.

Reply:
When we wrote the original ticket, we made a mistake about the dependencies between the test functions. I didn't realise that almost all the tests depended on generate. Sorry about that!

My top priority here is keeping most of the tests enabled.

Disabling tests means that we risk breaking Zebra's consensus-critical code, and only finding out much later. (Only after we finish the follow-up PR.)

We can achieve most of the modularity goals of this ticket by:
1. Leaving the existing Header::is_time_valid_at in zebra_chain
2. Leaving all the tests enabled
3. Adding a thin wrapper, also called is_time_valid_at, in zebra_consensus::block::check
4. Moving the tests that don't depend on generate to zebra_consensus

Then when we do #1021, we can move the code in Header::is_time_valid_at into zebra_consensus::block::check::is_time_valid_at, and move all the tests to zebra_consensus.

What do you think?

Reply:
@yaahc No issues.

@teor2345 Sure I can do that.